### PR TITLE
fix(sheets-ui): data-validation dropdown hidden when selection change

### PR DESCRIPTION
**Pull request by weird94**
close https://github.com/dream-num/univer-pro/issues/1724
close https://github.com/dream-num/univer-pro/issues/2088

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by wzhudev** [comment]
@hexf00 The bot failed to tag `qa:unverified`.

File: packages/sheets-ui/src/controllers/shortcuts/value.shortcut.ts [hunk 1]
```diff
@@ -27,3 +27,10 @@ export const ClearSelectionValueShortcutItem: IShortcutItem = {
     binding: KeyCode.DELETE,
     mac: KeyCode.BACKSPACE,
 };
+
+export const ClearSelectionValueBackspaceShortcutItem: IShortcutItem = {
+    id: ClearSelectionContentCommand.id,
+    // when focusing on any other input tag do not trigger this shortcut
+    preconditions: (contextService) => whenSheetEditorFocused(contextService),
+    binding: KeyCode.BACKSPACE,
```
Comment (wzhudev) [inline packages/sheets-ui/src/controllers/shortcuts/value.shortcut.ts:35]: In Excel, this behavior should be "start editing with an empty cell editor". The `ClearSelectionContentCommand` is carried by `KeyCode.DELETE`, and already implemented in the lines above.

**Review by wzhudev — CHANGES_REQUESTED** [review]
User behavior is not correct for Windows users.

**Comment by hexf00** [comment]
> @hexf00 The bot failed to tag `qa:unverified`.

Automatically add the `qa:untested` label if the PR title starts with `fix:`.

File: packages/sheets-thread-comment/src/controllers/sheets-thread-comment.controller.ts [hunk 2]
```diff
@@ -243,25 +246,29 @@ export class SheetsThreadCommentController extends Disposable {
...
+                        strokeWidth: 1,
+                        stroke: '#FFBD37',
+                        widgets: {},
+                    },
+                    primary: null,
                 },
-                primary: null,
-            });
+                [],
+                -1
```
Comment (wzhudev) [inline packages/sheets-thread-comment/src/controllers/sheets-thread-comment.controller.ts:270]: This selection should be marked as `evented: false` instead.

**Review by wzhudev — APPROVED** [review]

**Pull request merged**